File: problems/704-binary-search/binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int ans = -1, l = 0, r = size(nums) - 1;
        while(ans == -1 && l <= r){
            int mid = (l + r) / 2;
            if(nums[mid] == target) ans = mid;
            else if(nums[mid] < target) l = mid + 1;
            else r = mid - 1;
        }
        return ans;
    }
};

File: problems/704-binary-search/binary-search.kt
class Solution {
    fun search(nums: IntArray, target: Int): Int {
        var l = 0
        var r = nums.size - 1

        var i = 0
        while (l <= r){
            i = (r + l) / 2
            var cur = nums[i]
            when {
                cur > target -> { r = i - 1 }
                cur < target -> { l = i + 1 }
                cur == target -> { return i }
            }
        }

        return  -1
    }
}
